### PR TITLE
add util to build tripolar grid

### PR DESCRIPTION
**Pull request by raphaeldussin**
PR #223 uses data files that are too big, this PR adds code to build a simple (simplistic?) tripolar grid for testing purposes only. PR #223 could then use this function to generate its own tests with synthetic data from a tripolar grid to a regular grid using some type of field (e.g. sin(lat) )

### CODE REVIEW

**Comment by raphaeldussin** [comment]
@huard @aulemahal this implements the `simple_tripolar_grid` function whose purpose you've already figured out. The remaining routines are copied from a GFDL package rather than adding a new dependency for very little code. This will help solve issues regarding testing with data from tripolar grids.

File: xesmf/util.py [hunk 1]
```diff
@@ -210,3 +210,143 @@ def split_polygons_and_holes(polys):
...
+
+
+# Constants
+PI_180 = np.pi / 180.0
+_default_Re = 6371.0e3  # MIDAS
+HUGE = 1.0e30
+
+
+def simple_tripolar_grid(nlons, nlats, lat_cap=60, lon_cut=-300):
+    """generate a simplistic tripolar grid, regular under lat_cap
```
Comment (huard) [inline xesmf/util.py:222]:
```suggestion
    """Generate a simple tripolar grid, regular under `lat_cap`.
```

File: xesmf/util.py [hunk 2]
```diff
@@ -210,3 +210,143 @@ def split_polygons_and_holes(polys):
...
+
+    nlons: number of longitude points (int)
+    nlats: number of latitude points (int)
+    lat_cap: latitude of the northern cap (bipole)
+    lon_cut: longitude of the periodic boundary
```
Comment (huard) [inline xesmf/util.py:227]:
```suggestion
    Parameters
    ----------
    nlons: int
      Number of longitude points.
    nlats: int
      Number of latitude points.
    lat_cap: float 
      Latitude of the northern cap.
    lon_cut: float 
      Longitude of the periodic boundary.
```

**Review by huard — APPROVED** [review]
I suggest making `bipolar_projection` and `generate_bipolar_cap_mesh` private functions.

**Review by huard — APPROVED** [review]

**Comment by raphaeldussin** [comment]
@aulemahal friendly reminder

**Review by aulemahal — APPROVED** [review]
I do not have much knowledge of tripolar grids, but this code looks fine!

**Pull request merged**